src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -395,9 +395,9 @@ private void InitializeFilterList(UserSettings settings)
         private void PackageManagerLoaded(object sender, RoutedEventArgs e)
         {
             var timeSpan = GetTimeSinceLastRefreshAndRestart();
-            // Do not trigger a refresh if the browse tab is open and this is not the first load of the control.
+            // Do not trigger a refresh if this is not the first load of the control.
             // The loaded event is triggered once all the data binding has occurred, which effectively means we'll just display what was loaded earlier and not trigger another search
-            if (!(_loadedAndInitialized && _topPanel.Filter == ItemFilter.All))
+            if (!(_loadedAndInitialized))
```

Comment:
Isn't this fixing https://github.com/NuGet/Home/issues/4176? 

In this case, fixing by just not doing any refreshes by refocusing at all. 

The reason why we didn't do it for anything but browse was because every other tab can be changed by VS ambient state (for example, .NET Core projects can be updated through commandline, manually editing the csproj in VS or another editor.

Reply:
@nkolev92 the Loaded event of the PackageManagerControl won't fire in all the scenarios mentioned. Loaded doesn't fire when a change is made on the commandline, for example. I observed the Loaded event firing multiple times in a row when debugging a Console application and once when switching the UI to a code editor and back to the PMUI.

The refresh that occurs when the control is loaded was interfering with the selection of packages to update. There does not seem to be a predictable way to know when the control will be loaded and fire its Loaded event. For example, in the scenario where a code editor gets focused, the PMUI control does not have its Unloaded event fired. However, when the PMUI gets focus again after the code editor, it will fire the Loaded event.

Should we undo the change to not refresh update and installed? I could spend more effort on detecting when the control will fire the Loaded event and refresh with the Update and Installed tabs or I could make note of the effects the refresh on Loaded has on the feature in this PR that should be fixed with https://github.com/NuGet/Home/issues/4176 with an alternative auto-refresh mechanism.

Reply:
> @nkolev92 the Loaded event of the PackageManagerControl won't fire in all the scenarios mentioned. Loaded doesn't fire when a change is made on the commandline, for example. I observed the Loaded event firing multiple times in a row when debugging a Console application and once when switching the UI to a code editor and back to the PMUI.

I think my comment was misleading. 

The PM UI will refresh for many reasons. One of those reasons is focusing the PM UI (when Loaded should be called). 
Another reason is `Cache_Updated`, or basically a hand-edit of a PackageReference SDK based project.

A while back, NuGet would refresh on the CacheUpdated event *even* when the PM UI was not in focus. That's totally unnecessary, cause a refresh does *a lot of work*. 

https://github.com/NuGet/NuGet.Client/blob/b5b44526dea0379ebd6c8e51e8a041c06d5845ca/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L314-L323 is where that code is.

Today, the implementation just skips it when the package is not visible. 

> The refresh that occurs when the control is loaded was interfering with the selection of packages to update

This is the only part, that I'll say, I haven't spent enough time in this area lately to be able to understand why that's happening. 

> Should we undo the change to not refresh update and installed when the control's Loaded event is fired? I 

Given that this has a behavioral change, I think it's worth talking to PM.

Reply:
@JonDouglas we discussed offline that we will address this and not refresh on the control Loaded event when Installed and Updates tabs are selected.
I will address this and make a fix as a separate commit associated with the https://github.com/NuGet/Home/issues/4176 issue so we can monitor its affect and manage that code change separately from this feature.

Reply:
We're gonna change the behavior here & keep a commit to revert it if that behavior does not add more value than it removes (that we aren't sure of at the moment). We believe this is the right way forward given we've heard a number of complaints around the "refreshing behavior" as per #4176 and various customer interviews / survey verbatims.

Reply:
Created PR https://github.com/NuGet/NuGet.Client/pull/3813 to address this issue separately.